accepted/0000-prometheus-tls.md

```diff
@@ -0,0 +1,238 @@
+- Feature Name: Prometheus TLS support
+- Start Date: 2021-02-11
+- RFC PR:
+
+# Summary
+
+[summary]: #summary
+
+Secure connection to Prometheus server and exporters.
+
+# Motivation
+
+[motivation]: #motivation
+
+## Prometheus
+
+In the default configuration it is presumed that untrusted users have access to
+the Prometheus HTTP endpoint. They have access to all time series information
+contained in the database, plus a variety of operational/debugging
+information [^1].
+
+Applications needing access to Prometheus API:
+
+* _Grafana_
+* _Alertmanager_
+
+Users needing access to Prometheus API/UI:
+
+* monitoring administrator
+
+## Exporters
+
+Exporters generally only talk to one configured client instance with a preset
+set of commands/requests, which cannot be expanded via their HTTP endpoint.
+Still, all exposed metrics are available to untrusted users.
+
+> Uyuni should support and automate setting up secure configuration of
+> Prometheus and exporters.
+
+# Detailed design
+
+[design]: #detailed-design
+
+## Prometheus
+
+Prometheus 2.24.0 has introduced TLS encryption as well as basic and
+client-certificate authentication [^2]. All available configuration options are
+described in official documentation [^3]. Basic authentication credentials are
+stored in the configuration file. Passwords are hashed with bcrypt.
+
+To follow current SUSE Manager security model [^4] we should consider
+implementing following default configuration:
+
+* Upgrade Prometheus to version >= 2.24.1.
+* Create and deploy CA signed Prometheus server certificate.
+* Configure `tls_server_config` using this certificate.
+* Configure basic authentication accounts to map SUSE Manager credentials.
```

Comment:
Can you clarify the scope of this RFC more exactly? Looking at the title, I had the impression this was just about encryption, but here you are talking about authentication, leaving me to wonder whether the intention is actually to provide fully secured communication channels. Please adapt the one or the other :wink:

Reply:
Agree. Updated the summary.